src/tests/test_runtime_env.py

```diff
@@ -0,0 +1,550 @@
+            elif RUNTIME_ENV.IN_DENO:
+                load_script_type = "dynamic_import"
+            elif RUNTIME_ENV.IN_BUN:
+                load_script_type = "dynamic_import"
+            elif RUNTIME_ENV.IN_SHELL:
+                load_script_type = "load_function"
+            else:
+                # This should not happen - should throw error
+                load_script_type = "error"
+
+            # Should always determine a loading method
+            assert load_script_type is not None, "Should always determine a script loading method"
+
+            # Verify method is one of the expected options (error case means runtime detection failed)
+            expected_types = [
+                "dynamic_import", "importScripts_or_dynamic_import",
+                "nodeLoadScript", "load_function", "error"
+            ]
+            assert load_script_type in expected_types, f"Unexpected load script type: {load_script_type}"
+
+            # Should not be error type in normal cases
+            if not (RUNTIME_ENV.IN_NODE or RUNTIME_ENV.IN_DENO or RUNTIME_ENV.IN_BUN or
+                   RUNTIME_ENV.IN_BROWSER or RUNTIME_ENV.IN_SHELL):
+                assert load_script_type == "error", "Should error for unknown runtime"
+            else:
+                assert load_script_type != "error", "Should not error for known runtime"
+        """)
+
+    def test_compat_error_prevention(self, selenium):
+        """Test compat error prevention."""
+        selenium.run("""
+            from pyodide_js import RUNTIME_ENV
+
+            # Test that we can always determine the runtime environment
+            runtime_count = sum([
+                RUNTIME_ENV.IN_NODE,
+                RUNTIME_ENV.IN_DENO,
+                RUNTIME_ENV.IN_BUN,
+                RUNTIME_ENV.IN_BROWSER,
+                RUNTIME_ENV.IN_SHELL
+            ])
+
+            # Should be in at least one runtime environment
+            assert runtime_count >= 1, "Should be in at least one runtime environment"
+
+            # Test that the combination makes sense
+            if RUNTIME_ENV.IN_BROWSER:
+                # Browser should not be server runtime
+                server_count = sum([RUNTIME_ENV.IN_NODE, RUNTIME_ENV.IN_DENO, RUNTIME_ENV.IN_BUN])
+                assert server_count == 0, "Browser should not be server runtime"
+
+            # Test that we can determine appropriate functions for current runtime
+            has_appropriate_functions = True
+
+            # Every runtime should have appropriate compat functions
+            if RUNTIME_ENV.IN_NODE or RUNTIME_ENV.IN_DENO or RUNTIME_ENV.IN_BUN or RUNTIME_ENV.IN_BROWSER or RUNTIME_ENV.IN_SHELL:
+                has_appropriate_functions = True
+            else:
+                has_appropriate_functions = False
+
+            assert has_appropriate_functions, "Should have appropriate compat functions for current runtime"
+        """)
+
+
+class TestRuntimeEnvIntegration:
```

Comment:
This too. We don't need to test these internal behaviors, so I think we can remove it.